apiserver/plane/app/views/external/base.py

```
@@ -3,69 +1,165 @@
+    # If no model specified, use provider's default
+    if not model:
+        model = provider.default_model
+
+    # Validate model is supported by provider
+    if model not in provider.models:
+        return None, None, None
+
+    return api_key, model, provider_key
+
+
+def get_llm_response(task, prompt, api_key: str, model: str, provider: str) -> Tuple[str | None, str | None]:
```

Ruff (0.8.2)

Line too long (109 > 88)

(E501)